lib/counter_culture/with_connection.rb

```diff
@@ -6,22 +6,27 @@ def initialize(recipient)
 
     attr_reader :recipient
 
-    def call
+    def call(reading: false)
+      use_read_replica = CounterCulture.configuration.use_read_replica && reading
+      role = use_read_replica ? :reading : :writing
+
+      ActiveRecord::Base.connected_to(role: role) do
```

Comment:
I think to get the tests to pass for older Rails versions we'll have to do this only if we're at Rails 7.1 or greater, right?

Reply:
Yeah that's right!